code/game/machinery/vending.dm

```diff
@@ -94,6 +94,9 @@
 		build_inventory(premium, 0, 1, start_empty = 1)
 		for(var/obj/item/weapon/vending_refill/VR in component_parts)
 			refill_inventory(VR, product_records)
+		for(var/obj/item/weapon/vending_refill/VR in component_parts)
+			refill_inventory(VR, coin_records)
+			refill_inventory(VR, hidden_records)

```

Comment:
Why this needs to be in separate loop?
It seems to loop through exactly same things loop 2 lines above cycles through